src/sidebar/components/ThreadList.tsx

```diff
@@ -209,6 +209,18 @@ export default function ThreadList({ threads }: ThreadListProps) {
     }
   }, [store, newAnnotationTag]);
 
+  const editing = store.countDrafts() > 0;
+  const highlightedAnnotations = store.highlightedAnnotations();
+
+  // Scroll to the first highlighted annotation, unless creating/editing another
+  // annotation
+  useEffect(() => {
+    const [firstHighlightedAnnotation] = highlightedAnnotations;
+    if (!editing && firstHighlightedAnnotation) {
+      setScrollToId(firstHighlightedAnnotation);
+    }
+  }, [editing, highlightedAnnotations]);
```

Comment:
I started with an algorithm where scrolling happens to the first highlighted annotation, because it was a pretty simple logic to implement.